clients/src/main/java/org/apache/kafka/common/utils/ByteBufferOutputStream.java

```diff
@@ -105,7 +105,7 @@ public int initialCapacity() {
         return initialCapacity;
     }
 
-    private void maybeExpandBuffer(int remainingRequired) {
+    public void maybeExpandBuffer(int remainingRequired) {
```

Comment:
ArrayList has a public `ensureCapacity` and this looks similar (so there's precedent). Maybe we can use a similar naming convention?